queue/init.go

```diff
@@ -1,25 +1,48 @@
 package queue
 
-func Init() {
-	if theQueue != nil {
-		return
-	}
-	theQueue = make(chan *CommentQueue, N_COMMENT_QUEUE)
-	theQuit = make([]chan struct{}, N_COMMENT_QUEUE)
+import (
+	"context"
+	"encoding/json"
 
-	for idx := 0; idx < N_COMMENT_QUEUE; idx++ {
-		theQuit[idx] = make(chan struct{})
-		go ProcessCommentQueueLoop(idx, theQuit[idx])
-	}
-}
+	"github.com/golang-queue/queue"
+	"github.com/golang-queue/queue/simple"
+)
 
-func Close() {
-	for idx := 0; idx < N_COMMENT_QUEUE; idx++ {
-		theQuit[idx] <- struct{}{}
+var client *queue.Queue
+
+func Start() error {
+	var err error
+	// define the worker
+	w := simple.NewWorker(
+		simple.WithQueueNum(N_COMMENT_QUEUE),
+		simple.WithRunFunc(func(ctx context.Context, m queue.QueuedMessage) error {
+			v, ok := m.(*CommentQueue)
+			if !ok {
+				if err := json.Unmarshal(m.Bytes(), &v); err != nil {
```

Comment:
It's supposed to receive v as CommentQueue and should never goes to this code-block.

I'm ok with defensive-programming though～

Reply:
Found that the reason why we need this is because v is queue.Job, not *CommentQueue.

I also found that golang-queue intends to use json.Marshal to encode the task-body?
json.Marshal may not be good here because we may have 10k of the comments, 
resulting  possible 100k+ bytes in CommentQueue.CommentDBCS.

Is it possible that we can improve golang-queue and have queue.QueuedMessage as (*CommentQueue)?
Or we can have 

vJob, ok := m.(queue.Job)
v, ok := vJob.(*CommentQueue) ?
(no need to do json.Marshal / json.Unmarshal)

Reply:
I will take it and think about how to resolve the issue.

Reply:
Found that you are using the func object to work-around this issue.

I'll give the PR pass～

Reply:
Yes. golang-queue/queue supports two ways to handle worker model.

1. use any `func()` you want for single-machine model.
2. use Bytes() for distributed messaging like NSQ or NATs.